RealmSwift/Object.swift

```diff
@@ -266,17 +266,22 @@ open class Object: RLMObjectBase, ThreadConfined {
                                    to: List<DynamicObject>.self)
     }
 
-    // MARK: Equatable
-
+    // MARK: Comparison
     /**
-     Returns whether two Realm objects are equal.
+     Returns whether two Realm objects are the same.
 
-     Objects are considered equal if and only if they are both managed by the same Realm and point to the same
-     underlying object in the database.
+     Objects are considered the same if and only if they are both managed by the same
+     Realm and point to the same underlying object in the database.
+     
+     - note: Equality comparison is implemented by `isEqual(_:)`, but objects are only
+             considered equatable using `==` or `isEqual(_:)` if a primary key is
```

Comment:
Again, this comment doesn't seem to match the implementation. Objects without a primary key can be compared, they're just compared using object identity rather than checking if they refer to the same underlying object in the Realm.